In [2]:
import pandas as pd
import numpy as np

url = "https://raw.githubusercontent.com/angelarivass/Angela_MineriaDeDatos/refs/heads/main/googleplaystore.csv"
 
df = pd.read_csv(url)

In [3]:
df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [4]:
print("Tamaño del dataset: ", len(df))

Tamaño del dataset:  10841


A: 
Es arquitectónicamente seguro eliminar las filas de Android Ver porque esa cantidad de datos no afecta de manera significativa el análisis o los resultados. 
Por otro lado, si se eliminaran las filas de Rating se estarían eliminando casi 1500 registros de todo el dataset, lo que significa que se perderían una gran cantidad de datos (1,500 de 10,800).

In [5]:
df.dropna(subset=['Android Ver', 'Type'])


,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up
...,...,...,...,...,...,...,...,...,...,...,...,...,...
10836,Sya9a Maroc - FR,FAMILY,4.5,38,53M,"5,000+",Free,0,Everyone,Education,"July 25, 2017",1.48,4.1 and up
10837,Fr. Mike Schmitz Audio Teachings,FAMILY,5.0,4,3.6M,100+,Free,0,Everyone,Education,"July 6, 2018",1.0,4.1 and up
10838,Parkinson Exercices FR,MEDICAL,NaN,3,9.5M,"1,000+",Free,0,Everyone,Medical,"January 20, 2017",1.0,2.2 and up
10839,The SCP Foundation DB fr nn5n,BOOKS_AND_REFERENCE,4.5,114,Varies with device,"1,000+",Free,0,Mature 17+,Books & Reference,"January 19, 2015",Varies with device,Varies with device


B:
Como podemos observar, el tamaño del dataset apenas disminuyó:
Antes: 10841
Ahora: 10837

In [6]:
medianas = df.groupby('Category')['Rating'].transform('median')
df['Rating'] = df['Rating'].fillna(medianas)

print("Nulos en Rating:", df['Rating'].isnull().sum())

Nulos en Rating: 0


C:
Ya no hay nulos en los ratings.

D: Eliminando valor no numerico en categoría Size, asi se veía antes:

In [7]:
df['Size']

0                       19M
1                       14M
2                      8.7M
3                       25M
4                      2.8M
                ...        
10836                   53M
10837                  3.6M
10838                  9.5M
10839    Varies with device
10840                   19M
Name: Size, Length: 10841, dtype: object

Reemplazando "Varies with device" con Nan:

In [8]:
df['Size'] = df['Size'].replace("Varies with device", np.nan)

print(df['Size'])

0         19M
1         14M
2        8.7M
3         25M
4        2.8M
         ... 
10836     53M
10837    3.6M
10838    9.5M
10839     NaN
10840     19M
Name: Size, Length: 10841, dtype: object


In [9]:
print("Nulos en Size ahora:", df['Size'].isnull().sum())

Nulos en Size ahora: 1695


D:
Es mejor convertirlos a NaN ya que esos valores se excluyen de las funciones y si fueran cambiados por 0 sí serían considerado para calculos ya que el 0 sigue siendo un número, y afectaría por completo los resultados. Por ejemplo, si se sacara el promedio de los tamaños, los valores de 0 sesgarían el resultado hacia la izquierda. 

In [11]:

# Depuracion de duplicados
# df.duplicated(subset=['App'])
# Parse, castear un tipo de dato a otro (parseInt para cambiar a int)
# Tensor es un arreglo de datos